packages/core/src/zip-foundry.ts

```diff
@@ -0,0 +1,223 @@
+import { printContract } from "./print";
+import SOLIDITY_VERSION from './solidity-version.json';
+import contracts from '../openzeppelin-contracts';
+import { formatLinesWithSpaces, Lines, spaceBetween } from "./utils/format-lines";
+
+function getHeader(c: Contract) {
+  return [
+    `// SPDX-License-Identifier: ${c.license}`,
+    `pragma solidity ^${SOLIDITY_VERSION};`
+  ];
+}
+
+const test = (c: Contract, opts?: GenericOptions) => {
+  return formatLinesWithSpaces(
+    2,
+    ...spaceBetween(
+      getHeader(c),
+      getImports(c),
+      getTestCase(c),
+    ),
+  );
+
+  function getImports(c: Contract) {
+    return [
+      'import "forge-std/Test.sol";',
+      `import "../src/${c.name}.sol";`,
+    ];
+  }
+
+  function getTestCase(c: Contract) {
+    return [
+      `contract ${c.name}Test is Test {`,
+      spaceBetween(
+        [
+          `${c.name} public instance;`,
+        ],
+        [
+          'function setUp() public {',
+          [
+            `instance = new ${c.name}();`,
+          ],
+          '}',
+        ],
+        getContractSpecificTestFunction(),
+      ),
+      '}',
+    ];
+  }
+
+  function getContractSpecificTestFunction(): Lines[] {
+    if (opts !== undefined) {
+      switch (opts.kind) {
+        case 'ERC20':
+        case 'ERC721':
+          return [
+            'function testName() public {',
+            [
+              `assertEq(instance.name(), "${opts.name}");`
+            ],
+            '}',
+          ];
+
+        case 'ERC1155':
+          return [
+            'function testUri() public {',
+            [
+              `assertEq(instance.uri(0), "${opts.uri}");`
+            ],
+            '}',
+          ];
+
+        case 'Governor':
+        case 'Custom':
+          return [
+            'function testSomething() public {',
+            [
+              '// Add your test here',
+            ],
+            '}',
+          ]
+
+        default:
+          throw new Error('Unknown ERC');
+      }
+    }
+    return [];
+  }
+};
+
+const script = (c: Contract) => {
+  return formatLinesWithSpaces(
+    2,
+    ...spaceBetween(
+      getHeader(c),
+      getImports(c),
+      getScript(c),
+    ),
+  );
+
+  function getImports(c: Contract) {
+    return [
+      'import "forge-std/Script.sol";',
+      `import "../src/${c.name}.sol";`,
+    ];
+  }
+
+  function getScript(c: Contract) {
+    return [
+      `contract ${c.name}Script is Script {`,
+      spaceBetween(
+        [
+          'function setUp() public {}',
+        ],
+        [
+          'function run() public {',
+          [
+            'vm.startBroadcast();',
+            `${c.name} instance = new ${c.name}();`,
+            'console.log("Contract deployed to %s", address(instance));',
+            'vm.stopBroadcast();',
+          ],
+          '}',
+        ],
+      ),
+      '}',
+    ];
+  }
+};
+
+const setupSh = `\
+# Check if git is installed
+if ! [ -x "$(command -v git)" ]
```

Comment:
```suggestion
if ! command -v git &> /dev/null
```

Reply:
This didn't work on the Github Action when I tried it.  See https://github.com/OpenZeppelin/contracts-wizard/actions/runs/5536781265/jobs/10104887999 -- it somehow went into the if-statement to print `git command not found. Install git and try again.` but also printed the git path `/usr/bin/git`

Reply:
It may be because `command` is a builtin, not sure though.

We could instead use `which` (it's what I would have used initially).

```suggestion
if ! which git &> /dev/null
```

Reply:
Changed to use `which`